js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -1515,51 +1525,70 @@ const EditingController = modules.ViewController.inherit((function() {
             }
         },
         _saveEditDataCore: function(deferreds, results, changes) {
-            const that = this;
-            const store = that._dataController.store();
-            let isDataSaved = true;
+            const onSavingParams = {
+                cancel: false,
+                promise: null
+            };
+            this.executeAction('onSaving', onSavingParams);
+            const d = new Deferred();
+            when(fromPromise(onSavingParams.promise))
```

Comment:
Yes, it is possible, but 2 separate parameters are better because:
1. 1 parameter - 1 type, it is easier for TS and looks like _true_ static type system. 
2. Code like `e.cancel = new Promise((resolve) => { resolve(true) })` is not obvious. 
3. These parameters accomplish different tasks

We don't have many parameters in this callback so I think it's OK.